backend/src/database/state-migrations/8-treatment-system-improvements.ts

```diff
@@ -0,0 +1,118 @@
+
+export const treatmentSystemImprovements8: Migration = {
+    actions: (_initialState, actions: any[]) => {
+        for (const action of actions) {
+            switch (action?.type) {
+                case '[Vehicle] Add vehicle':
+                    for (const material of action.materials) {
+                        migrateMaterial(material);
+                    }
+                    for (const personnel of action.personnel) {
+                        migratePersonnel(personnel);
+                    }
+                    break;
+                case '[Patient] Add patient':
+                    migratePatient(action.patient);
+                    break;
+            }
+        }
+    },
+    state: (state: any) => {
+        // Spatial tree
+        state.spatialTrees = cloneDeepMutable({
+            materials: SpatialTree.create(),
+            patients: SpatialTree.create(),
+            personnel: SpatialTree.create(),
+        });
+        for (const material of StrictObject.values(state.materials)) {
+            migrateMaterial(material);
+            if (!material.position) {
+                continue;
+            }
+            SpatialTree.addElement(
+                state.spatialTrees.materials,
+                material.id,
+                material.position
+            );
+        }
+        for (const personnel of StrictObject.values(state.personnel)) {
+            migratePersonnel(personnel);
+            if (!personnel.position) {
+                continue;
+            }
+            SpatialTree.addElement(
+                state.spatialTrees.personnel,
+                personnel.id,
+                personnel.position
+            );
+        }
+        // Migrate patients
+        for (const patient of StrictObject.values(state.patients)) {
+            migratePatient(patient);
+            for (const personnel of StrictObject.values(state.personnel)) {
+                if (personnel.assignedPatientIds[patient.id]) {
+                    patient.assignedPersonnelIds[personnel.id] = true;
+                }
+            }
+            for (const material of StrictObject.values(state.materials)) {
+                if (material.assignedPatientIds[patient.id]) {
+                    patient.assignedMaterialIds[material.id] = true;
+                }
+            }
+            if (!patient.position) {
+                continue;
+            }
+            SpatialTree.addElement(
+                state.spatialTrees.patients,
+                patient.id,
+                patient.position
+            );
+        }
+        // VehicleTemplates
+        for (const vehicleTemplate of state.vehicleTemplates) {
+            vehicleTemplate.materials = vehicleTemplate.materials.map(
+                () => 'standard'
+            );
+        }
+        // PatientCategories
+        for (const patientCategory of state.patientCategories as Mutable<
+            ExerciseState['patientCategories']
```

Comment:
This is not safe to do at this point. You cannot assume anything about the state because it may have changed in later versions. When doing it like this you have to change this section every time this part of the state changes. Are you aware of that?

Reply:
I would interpret this a bit different:

"You can't use this typing here, as the State type could change in later versions. You would notice this if `ExerciseState['patientCategories']` changes because then there would be a type error."

The same also goes for the `Mutable<ImageProperties>` at the end of this file.

Reply:
Yes, this was what I meant. You can leave it like this, but you must be aware that this might break at some point.